models/chapter_fetch.go

```diff
@@ -95,9 +102,9 @@ func (c *Chapter) FetchImageURLs(ctx context.Context) (imgUrls []*url.URL, err e
 	}
 
 	// Pre-allocate the slice
-	imgUrls = make([]*url.URL, 0, len(imgStrs))
+	imgUrls = make([]*url.URL, len(imgStrs), len(imgStrs))
```

Comment:
This change to the `make` and the for loop is unnecessary. The array is made with the capacity, so `append` will not cause a re-allocation.
Doing it with `append` means that `imgUrls` never contains null pointers.